### PR TITLE
Port FunctionalTests.Transitional to .NET Core 3

### PR DESCRIPTION
**Pull request by bricelam**
Part of #271

### CODE REVIEW

File: src/EntityFramework/Infrastructure/DependencyResolution/DefaultProviderFactoryResolver.cs [hunk 1]
```diff
@@ -31,6 +32,11 @@ private static object GetService(Type type, object key, Func<ArgumentException,
                 }
                 catch (ArgumentException e)
                 {
+                    if (string.Equals(name, "System.Data.SqlClient", StringComparison.OrdinalIgnoreCase))
+                    {
+                        return SqlClientFactory.Instance;
```
Comment (bricelam) [inline src/EntityFramework/Infrastructure/DependencyResolution/DefaultProviderFactoryResolver.cs]: @ajcvickers @divega If we don't find a factory for SqlClient, we'll return this hard-coded one.

File: src/EntityFramework/Utilities/DbProviderFactoryExtensions.cs [hunk 2]
```diff
@@ -29,6 +30,11 @@ public static string GetProviderInvariantName(this DbProviderFactory factory)
 
             if (row == null)
             {
+                if (factory is SqlClientFactory)
+                {
+                    return "System.Data.SqlClient";
```
Comment (bricelam) [inline src/EntityFramework/Utilities/DbProviderFactoryExtensions.cs]: Ditto for mapping the factory to its invariant name.

File: src/EntityFramework/Infrastructure/DependencyResolution/DefaultProviderFactoryResolver.cs (hunk 1 above)
Comment (divega) [reply src/EntityFramework/Infrastructure/DependencyResolution/DefaultProviderFactoryResolver.cs]: I think this is fine. It should fix the usability issues, but I am not sure about the full implications. Maybe check it in but then talk about it a bit in the design meeting. It also has some implications for  https://github.com/aspnet/EntityFramework6/issues/823 (e.g. using Microsoft.Data.SqlClient).

File: src/EntityFramework/Utilities/DbProviderFactoryExtensions.cs (hunk 2 above)
Comment (divega) [reply src/EntityFramework/Utilities/DbProviderFactoryExtensions.cs]: Same as above.

File: src/EntityFramework/Infrastructure/DependencyResolution/DefaultProviderFactoryResolver.cs (hunk 1 above)
Comment (ajcvickers) [reply src/EntityFramework/Infrastructure/DependencyResolution/DefaultProviderFactoryResolver.cs]: Agreed that this is fine but we need to talk about #823. I have some ideas.

**Pull request merged**